GATEWAY/src/providers/devices.service.ts

```diff
@@ -7,6 +7,8 @@ import { Device } from './utils.service';
 @Injectable()
 export class DevicesService {
   public devices: Device[];
+  public flagThen: boolean = false;
+  public flagCatch: boolean = false;
```

Comment:
Actualle need these flags, as they are used to differentiate between if `convertBLEDeviceToDevice` returns a device through a `.then` or a `.catch`.
This is because they return the same object, and are impossible to se the difference between them.